web/src/providers/CreateTest/CreateTest.provider.tsx

```diff
@@ -47,10 +47,34 @@ const CreateTestProvider = ({children}: IProps) => {
   const stepNumber = useAppSelector(CreateTestSelectors.selectStepNumber);
   const pluginName = useAppSelector(CreateTestSelectors.selectPlugin);
   const activeStep = useAppSelector(CreateTestSelectors.selectActiveStep);
+  const isFinalStep = stepNumber === stepList.length - 1;
 
-  const onNext = useCallback(() => {
-    dispatch(setStepNumber({stepNumber: stepNumber + 1}));
-  }, [dispatch, stepNumber]);
+  const onCreateTest = useCallback(
+    async (draft: TDraftTest) => {
+      const test = await createTest(draft).unwrap();
+      const run = await runTest({testId: test.id}).unwrap();
+
+      navigate(`/test/${test.id}/run/${run.id}`);
+    },
+    [createTest, navigate, runTest]
+  );
+
+  const onUpdateDraftTest = useCallback(
+    (update: TDraftTest) => {
+      dispatch(setDraftTest({draftTest: update}));
+    },
+    [dispatch]
+  );
+
+  const onNext = useCallback(
```

Comment:
Converting this into a single atomic process for creating a test or updating the draft.